internal/util/failover.go

```diff
@@ -137,20 +137,34 @@ func GetPod(clientset kubernetes.Interface, deploymentName, namespace string) (*
 // By default information is only returned for replicas within the cluster.  However,
 // if primary information is also needed, the inlcudePrimary flag can set set to true
 // and primary information will will also be included in the ReplicationStatusResponse.
-func ReplicationStatus(request ReplicationStatusRequest, includePrimary bool) (ReplicationStatusResponse, error) {
+//
+// Also by default we do not include any "busted" Pods, e.g. a Pod that is not
+// in a happy phase. That Pod may be lacking a "role" label. From there, we zero
+// out the statistics and apply an error
+func ReplicationStatus(request ReplicationStatusRequest, includePrimary, includeBusted bool) (ReplicationStatusResponse, error) {
 	response := ReplicationStatusResponse{
 		Instances: make([]InstanceReplicationInfo, 0),
 	}
 
-	// First, get replica pods using selector pg-cluster=clusterName,role=replica if not including the primary,
-	// or pg-cluster=clusterName,pg-database if including the primary
-	var roleSelector string
+	// Build up the selector. First, create the base, which restricts to the
+	// current cluster
+	// pg-cluster=clusterName,pgo-pg-database
+	selector := fmt.Sprintf("%s=%s,%s",
+		config.LABEL_PG_CLUSTER, request.ClusterName, config.LABEL_PG_DATABASE)
+
+	// if we are not including the primary, determine if we are including busted
+	// replicas or not
 	if !includePrimary {
-		roleSelector = fmt.Sprintf("%s=%s", config.LABEL_PGHA_ROLE, config.LABEL_PGHA_ROLE_REPLICA)
-	} else {
-		roleSelector = config.LABEL_PG_DATABASE
+		if includeBusted {
+			// include all Pods that identify as a database, but **not** a primary
+			// pg-cluster=clusterName,pgo-pg-database,role!=config.LABEL_PGHA_ROLE_PRIMARY
```

Comment:
I think having something like `selector + role!=config.LABEL_PGHA_ROLE_PRIMARY` would make it more clear what this line is doing

Reply:
This shows the full selector string though.